CRM/Core/Payment/AuthorizeNet.php

```diff
@@ -133,10 +133,7 @@ public function doDirectPayment(&$params) {
     }
 
     if (!empty($params['is_recur']) && !empty($params['contributionRecurID'])) {
-      $result = $this->doRecurPayment();
-      if (is_a($result, 'CRM_Core_Error')) {
-        return $result;
-      }
+      $this->doRecurPayment();
       return $params;
```

Comment:
won't calling functions potentially miss error messages?

Reply:
doRecurPayment won't return an error now - it will throw exceptions